### ISSUE TITLE
Improperly Inserting Array Properties Into MongoDB

### ISSUE THREAD

**Issue by kevdougful** [strongloop/loopback-next#1508]
## Description / Steps to reproduce / Feature proposal
I have a `Publication` model that basically describes an entry in a bibliography.  I'd like to store the author names for each publication as an array of strings:
```ts
export class Publication extends Entity {
  // ...
  @property.array(String) authors: string[]
  // ...
}
```
and a controller like this:
```ts
export class PublicationController {

  constructor(@repository(PublicationRepository) protected repo: PublicationRepository) {}

  @post('/publications')
  async createPublication(@requestBody() publication: Publication) {
    return await this.repo.create(publication)
  }
}
```

## Current Behavior
I'm using MongoDB as a data source and the data is not being stored as an array but rather as a single comma-separated string.  I've found that if I comment out the `authors` property in the model, `POST` a new publication, then turn the `authors` property back on any subsequent `GET /publications` will return objects where the `authors` property is an array.

## Expected Behavior
I expect the `@property.array(String)` to properly insert data as an array in both the database and in responses from the REST API.

_See [Reporting Issues](http://loopback.io/doc/en/contrib/Reporting-issues.html) for more tips on writing good issues_

**Comment by jannyHou** [comment]
Trying it, will update my finding later.

**Comment by jannyHou** [comment]
@dhmlau Verified it's a bug. It needs time to investigate including some debug setup, I agree to put it in the backlog, changed the pipeline.

**Comment by shimks** [comment]
Fixed with #1570. Closing as done
